### DISCUSSION TITLE
reading how to makes no sense- integrate U8g2_for_Adafruit_GFX into tft_espi for my TFt.

### DISCUSSION THREAD

**Discussion by geetee24** [Bodmer/TFT_eSPI#2087] [Q&A - Fonts]
all i want is to use their 72 pixel font.

Reading the docs for U8g2_for_Adafruit_GFX  didnt answer that.

**Comment by Bodmer** [comment]
I can see you are struggling with the apparent complexity of the task. The docs will make more sense after running the examples of the library on the TFT and examining the comments in the sketch.

I suggest you load this library:
https://github.com/Bodmer/U8g2_for_TFT_eSPI

Bear in mind this is a port provided as a users request only, and that I do not provide support for it.

Use the "Code" button to download the library and then install in your IDE. There are online tutorials on how to do this if you are not sure so I will not repeat those instructions.

Run the U8g2_for_TFT_eSPI library example "Hello"

You will see [this line](https://github.com/Bodmer/U8g2_for_TFT_eSPI/blob/master/examples/Hello/Hello.ino#L39) loads a 14 pixel helvetica style font. In the comments [there is a link](https://github.com/olikraus/u8g2/wiki/fntlistall) to a web page where all the fonts and sizes available are listed.

Similarly [this line](https://github.com/Bodmer/U8g2_for_TFT_eSPI/blob/master/examples/Hello/Hello.ino#L39) loads a 63 pixel font.

Large fonts in the 17-99 pixel height range are listed [here](https://github.com/olikraus/u8g2/wiki/fntlist99).  For example to load the 78 pixel font you would use:
u8f.setFont(u8g2_font_logisoso78_tn);

Clearly you will be constrained in styles unless you generate your own fonts for that library (which I do not provide). Beware that proportioanl fonts are bad news as a narrow 1 will not fully overwrite a wider character like a 4.  The features built into TFT_eSPI fonts to overcome this are not implemented in that extension library.

TFT_eSPI is quite an advanced library for a beginner so you may be better off trying to use Adafruit_GFX as there is a [tutorial here](https://learn.adafruit.com/adafruit-gfx-graphics-library). TFT_eSPI supports some of the Adafruit_GFX functions but has diverged from full compatibility.

Without understanding your requirements better it is dificult to advise on the best approach.

Don't raise Q&A for help in the issues section, that section is to report potential software bugs in the library, not Q&A. The discussions area is for questions.

**Comment by Bodmer** [comment]
If you are updating changing values the the library here is probably a better option:
https://github.com/Bodmer/TFT_eSPI_ext

As noted in the ReadMe for that library: there are 3000 fonts available.
